docs/dev/infrastructure_as_code.rst

```diff
@@ -0,0 +1,39 @@
+===============================================================================
+Infrastructure as Code
+===============================================================================
+
+-------------------------------------------------------------------------------
+Overview
+-------------------------------------------------------------------------------
+
+We use `terraform <https://developer.hashicorp.com/terraform>`__ to manage some
+of our infrastructure. It lets us repeat infrastructure setup tasks without
+having to rely on individual developers clicking the right buttons in the right
+order.
```

Comment:
Will terrafrom break if we make changes to this infra in the UI?

Reply:
Terraform will tell you "hey, this looks different from what I expect" and refuse to make any new changes. It will give you some pointers as to what's different so you can change the terraform code to no longer be surprised by the manual changes.

If you delete something, you'll need to run an extra step - `terraform state rm <name-of-resource>` will cause terraform to forget about the resource entirely.

Reply:
Cool! Thanks for explaining.